components/Form/Form.tsx

```diff
@@ -32,15 +42,21 @@ const Form: React.FC<{
               <View style={style.field}>
                 <Input
                   placeholder={field.placeholder}
-                  onChangeText={val => onChange(val)}
+                  onChangeText={val => {
+                    const currentVal =
+                      field.keyboardType === 'numeric'
+                        ? val.replace(/[^0-9]/g, '')
+                        : val;
+                    onChange(currentVal);
```

Comment:
You can move this logic in a function to improve the readability. `onChangePlaceholderText`